blockprotocol/blockprotocol
# Handle WordPress editor being iFramed by Gutenberg plugin

## 🌟 What is the purpose of this PR?

Handles instances where the Gutenberg editor canvas is inside an iFrame, which causes issues with loading styles, explained in the doc comments

This fixes styles for blocks which use libraries which work by appending styles to the document head when reached via a global (e.g. `window`, `document`, `globalThis`), e.g. Webpack's `style-loader`, `emotion` (used by MUI), `stitch` (used by TLDraw).

It also makes the necessary version + changelog changes for `0.0.5`, which will be released from this code.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778935/1204225158472266/f) _(internal)_

## 🔍 What does this change?

- Changes are documented in the diff.

## 📜 Does this require a change to the docs?

- If we cannot eliminate all issues with the iFrame we may need to add more warnings in the WordPress plugin when the iFrame is encountered.

## ⚠️ Known issues
With the Gutenberg plugin active:
- custom elements do not load in the editor view – see comments in diff. Users will see a warning message if they attempt to use a custom element block.
- the Table block does not function properly (cannot add rows) 

## 🐾 Next steps

- Release a new version of the WordPress plugin with whatever fixes are available – the current fixes are enough of an improvement to warrant getting a new version out, even if the oustanding issue cannot be fixed.

## ❓ How to test this?



1.  Run WordPress locally as per the README in `libs/wordpress-plugin`
2. Install the Gutenberg plugin via the plugins setting page
3. Activate the Gutenberg plugin
4. add a bunch of blocks to a page, check they work properly (subject to exceptions mentioned under 'Known issues')
5. Deactivate the Gutenberg plugin, check everything still works
